core-addon/Core.js

```diff
@@ -521,6 +533,10 @@ export default class Core {
     // Clear the list of studies user took part in.
     await this._storage.clearActivatedStudies();
 
+    // Flip upload enabled to disabled: this will trigger a
+    // deletion-request.
+    Glean.setUploadEnabled(false);
```

Comment:
Just a general note for glean I guess, but `setUploadEnabled` generating a deletion ping feels like an unexpected side-effect :)

Reply:
> 
> 
> Just a general note for glean I guess, but `setUploadEnabled` generating a deletion ping feels like an unexpected side-effect :)

Heh, that's [intended](https://mozilla.github.io/glean/book/user/general-api.html#enabling-and-disabling-metrics): when disabling upload Glean respects the user intent by signaling pipeline to delete data. The Glean-owned deletion ping has a special treatment that allows it to go through even with upload disabled!